docs/next_link_analysis.md

```diff
@@ -0,0 +1,108 @@
+# Investigating relationships between successive URLs
+
+## Predictable relationships between consecutive URLs 
+
+The hypothesis is that there is no relationship between the sequences of government GOVUK website URLs visited by a user. Clicks are essentially randomly distributed, according to their availability. There is a natural degree of accuracy, of a naive prediction, just due the layout of the government website.
+
+In summary:
+
+- there is a [65%] chance that the next URL will be found in the next [10] recommended URLs
+- the Mean Reciprocal Rank (MRR) for the predictions is [0.43] on a holdout test set
+
+These results should not be regarded in terms of good or bad, rather they just establish a reference score. This serves as a benchmark to be beaten in future tailored predictions, which incorporate additional explanatory data regarding the user. This enables meaningful discussion and development of future models. These measures should be seen as the naiive or random prediction threshold. Any future models should be better than this measure.  
+
+## Data used
+
+The data used in the analysis is pulled from a single hour of usage (midday) of the GovUK website on 2020-11-13. This contains 844k unique records. As can be seen beneath, for a single `fullVisitorId`, there can be multiple values for `visitId`. It relates to hits which are classified as `PAGE`.
+
+## Illustration of data
+
+![visitor data](./images/visit_data_example.jpeg)
+
+As seen, pagePath represents the URL visited.
+
+## Rationale for considering only the preceding website
+
+Typically in time series processes, the most important data point is the current one, in determining the next data point. In a memoryless process, the only data point which is relevant is the current one and the following data point will be a random jump to one of the connected points in the space. In the context of the GovUK website, it is frequently the case that a visitor can only visit the next URL via the current URL.
```

Comment:
### Draft (might update this as I read further into the report)
This makes a lot of sense! Given that a visitor can only visit the next URL via the current URL, and they are offered a limited number of possible URLs to click on, then we could actually do something like a random selection and compare that with the actual clicks users have made to get an even more basic benchmark. *e.g. this gives us the accuracy if we just randomly guessed what people will click next*